File: app/src/main/java/com/youer/genshin/view/CharacterInfoView.java
package com.youer.genshin.view;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;
import com.youer.genshin.constants.Constants;
import com.youer.genshin.databinding.ViewCharacterInfoBinding;
import com.youer.genshin.req.CalculateReq.CharacterInfo;

/**
 * @author youer
 * @date 9/27/23
 */
public class CharacterInfoView extends LinearLayout {
    ViewCharacterInfoBinding binding;
    private CharacterInfo characterInfo;
    private DeleteListener listener;

    public CharacterInfoView(Context context, CharacterInfo characterInfo) {
        super(context);
        this.characterInfo = characterInfo;
        initView();
    }

    public CharacterInfoView setListener(DeleteListener listener) {
        this.listener = listener;
        return this;
    }

    private void initView() {
        binding = ViewCharacterInfoBinding.inflate(LayoutInflater.from(getContext()), this, true);
        binding.characterInfo.setText(
            Constants.CHARACTERS.get(characterInfo.getCharacterId()).getAsString()
                + "/"
                + Constants.RELICS.get(characterInfo.getGroupType()).getAsString());
        binding.delete.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                if (listener != null) {
                    listener.onDelete(CharacterInfoView.this, characterInfo);
                }
            }
        });
    }

    public interface DeleteListener {
        void onDelete(View view, CharacterInfo characterInfo);
    }

}

File: app/src/main/java/com/youer/genshin/view/CharacterInfoView.kt
package com.youer.genshin.view

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import com.youer.genshin.constants.Constants
import com.youer.genshin.databinding.ViewCharacterInfoBinding
import com.youer.genshin.req.CalculateReq.CharacterInfo
import com.youer.genshin.view.CharacterInfoView

/**
 * @author youer
 * @date 9/27/23
 */
class CharacterInfoView(context: Context, private val characterInfo: CharacterInfo) : LinearLayout(context) {
    private var binding: ViewCharacterInfoBinding
    var listener: DeleteListener? = null

    init {
        binding = ViewCharacterInfoBinding.inflate(LayoutInflater.from(context), this, true)
                .apply {
                    character.text = Constants.CHARACTERS?.let {
                        it[characterInfo.characterId].asString
                    } + "/" + Constants.RELICS?.let {
                        it[characterInfo.groupType].asString
                    }
                    delete.setOnClickListener {
                        listener?.onDelete(this@CharacterInfoView, characterInfo)
                    }


                }
    }

    interface DeleteListener {
        fun onDelete(view: View, characterInfo: CharacterInfo)
    }

}